std/stdio.d

```diff
@@ -2808,6 +2808,90 @@ is empty, throws an `Exception`. In case of an I/O error throws
         assert(i == witness.length);
     }
 
+/**
+Like `asCharsRange` but returns a range o `ubyte`s, not `char`s.
+*/
+    auto asUbytesRange(size_t bufferSize = 4096)
+    {
+        assert(bufferSize > 0);
+
+        import std.algorithm.iteration;
+        return byChunk(bufferSize).joiner;
+    }
+
+/**
+Returns an $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
+set up to read from the file handle a character at a time.
+
+The element type for the range will be `char`. Range primitives
```

Comment:
Ranges with elements of type `char` are a bit unusual due to autodecoding. Strings are decoded as ranges of `dchar`.

Reply:
Without ranges with elements of type `char`, it would be impossible to use functions such as `.byDchar` from `std.utf`. So, it is necessary.

If you want to change the API, first change the API of `std.utf` to use `ubyte` instead of `char`. My design is required by compatibility with `std.utf`.

Reply:
There is an ongoing effort to try to make autodecoding `nothrow`. The way this is going to work is that all `char` arrays / ranges are assumed to form valid UTF-8, so `char` arrays / ranges must be validated on creation. A standard library construct which produces `char` arrays / ranges without validation would thus break that contract.

Unvalidated characters, or characters in other encodings, are represented an `ubyte` arrays / ranges.